hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/CloudInfoCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.util.phonehome;
+
+import com.hazelcast.instance.impl.Node;
+
+import java.io.File;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.internal.util.EmptyStatement.ignore;
+
+public class CloudInfoCollector implements MetricsCollector {
+    private static final int TIMEOUT = 1000;
+    private static final int RESPONSE_OK = 200;
+
+    private static final Path KUBERNETES_TOKEN_PATH = Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/token");
+    private static final Path DOCKER_FILE_PATH = Paths.get("/.dockerenv");
+    private String awsEndPoint;
+    private String azureEndPoint;
+    private String gcpEndPoint;
+
+    public CloudInfoCollector(String awsEndpoint, String azureEndpoint, String gcpEndpoint) {
+        awsEndPoint = awsEndpoint;
+        azureEndPoint = azureEndpoint;
+        gcpEndPoint = gcpEndpoint;
+    }
+
+    public void modifyEndPoints(String awsEndpoint, String azureEndpoint, String gcpEndpoint) {
+        awsEndPoint = awsEndpoint;
+        azureEndPoint = azureEndpoint;
+        gcpEndPoint = gcpEndpoint;
+    }
+
+    public Map<PhoneHomeMetrics, String> computeMetrics(Node hazelcastNode) {
+
+        Map<PhoneHomeMetrics, String> environmentInfo = new HashMap<>();
+
+        if (check(awsEndPoint)) {
+            environmentInfo.put(PhoneHomeMetrics.CLOUD, "A");
+        } else if (check(azureEndPoint)) {
+            environmentInfo.put(PhoneHomeMetrics.CLOUD, "Z");
+        } else if (check(gcpEndPoint)) {
+            environmentInfo.put(PhoneHomeMetrics.CLOUD, "G");
+        } else {
+            environmentInfo.put(PhoneHomeMetrics.CLOUD, "-1");
+        }
+
+
+        File dockerFile =DOCKER_FILE_PATH.toFile();
+        if (dockerFile.exists()) {
+            File kubernetesToken = KUBERNETES_TOKEN_PATH.toFile();
```

Comment:
I think we can get by here without calling `toFile()` so that you don't have to go into that cumbersome mocking with PowerMockito. `Path` has a `toRealPath()` method which throws an `IOException` if the file doesn't exist. You can catch and use that exception.

Same for the `DOCKER_FILE_PATH` existence check.